.github/workflows/snapshoty.yml

```diff
@@ -0,0 +1,27 @@
+---
+
+name: snapshoty
+on:
+  workflow_run:
+    workflows:
+      - macos
+    types:
+      - completed
+
+permissions:
+  contents: read
+
+jobs:
+  macos-publish-snapshots:
```

Comment:
This will even run if the `macos` workflow failed, so, those artifacts won't be there in that particular case, hence you might need to use a `if` to filter if the workflow run completed successfully.